app/components/channel_drawer/channels_list/list/list.js

```diff
@@ -286,6 +183,16 @@ class List extends Component {
         });
     });
 
+    onSelectChannel = (channel, currentChannelId) => {
+        const {onSelectChannel} = this.props;
+        onSelectChannel(channel, currentChannelId);
+    };
+
+    onLayout = (event) => {
+        const {width} = event.nativeEvent.layout;
+        EventEmitter.emit(SET_WITDH, width - 40);
```

Comment:
Why isn't this done through state?

Reply:
cause doing it in the state is causing renderItems to re-create the rows and there's no need for that

Reply:
Why would it be doing that though? The SectionList shouldn't be redrawing if its props don't change